### PR TITLE
Set same env variables as sensor process and s/_sensor_service/sensor_service

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Fixes https://github.com/StackStorm/st2/issues/2694 and https://github.com/StackStorm/st2/issues/2693

Need feedback on https://github.com/StackStorm/st2docs/issues/156

### CODE REVIEW

**Comment by lakshmi-kannan** [comment]
Docs: https://github.com/StackStorm/st2docs/pull/157

File: st2actions/st2actions/runners/__init__.py [hunk 1]
```diff
@@ -136,9 +138,13 @@ def _get_common_action_env_variables(self):
         result['ST2_ACTION_PACK_NAME'] = self.get_pack_name()
```
Comment (Kami) [inline st2actions/st2actions/runners/__init__.py]: Let's please move this to a different place.

As I noted here https://github.com/StackStorm/st2/issues/2693#issuecomment-220262497, there are two different things:

1. Environment variables which are set for st2client to work (as mentioned, that's a hack, I don't like it but that's limitation of st2client right now)
2. Environment variables which are set as a convenience and available to actions.

In this case we are fixing 1), but you are doing in in code for 2).

File: st2reactor/st2reactor/sensor/base.py [hunk 2]
```diff
@@ -23,8 +23,10 @@ def __init__(self, sensor_service, config=None):
         :keyword config: Sensor config.
         :type config: ``dict`` or None
         """
-        self._sensor_service = sensor_service
-        self._config = config or {}
+        self._sensor_service = sensor_service  # Deprecate in the future
+        self.sensor_service = sensor_service
+        self._config = config or {}  # Deprecate in the future
+        self.config = self._config
```
Comment (Kami) [inline st2reactor/st2reactor/sensor/base.py]: I originally missed this PR (I blame thunderbird sync!) and that's how I've done it utilizing @property - https://github.com/StackStorm/st2/pull/2698/files#diff-f120a0125bcd3ec86f1d85020ed530ffR26

**Comment by Kami** [comment]
Let's please get this addressed - https://github.com/StackStorm/st2/pull/2695#discussion_r63854788

Besides that, LGTM, :+1:

**Comment by Kami** [comment]
As mentioned in #2699 (yes, that PR I opened because I missed this one), it would also be good to add some (integration / end to end) tests which test datastore operations to st2tests or similar.

The unit tests which we have right now only test implementation details - they test correct environment variables are set. What we really are interested in and should be testing is "do datastore related methods" work. For that, we need end to end tests.

**Comment by Kami** [comment]
Minor comment wrt to PR title and description - imo, it's better if the PR title includes a more high-level description of what is being fixed / changed. This makes it easier for anyone checking the PR to understand what is going on and what is being changed.

Right now it includes implementation detail ("setting same environment variables"and  by itself this doesn't tell much). Imo, a better title would be something along the lines of "Fix datastore access in the Python runner actions" or something along those lines since that what we are doing - we are fixing datastore access for Python runner actions (and "setting same environment variables" is just an implementation detail of the fix which can be included in the description).

**Comment by lakshmi-kannan** [comment]
As per slack discussion, here is an integration test for datastore access in action - https://github.com/StackStorm/st2tests/pull/47

**Comment by Kami** [comment]
Thanks for moving the code and adding a test :+1:

**Pull request merged**